### PR TITLE
Add ChainModel

### PR DESCRIPTION
**Pull request by promag**
Based on #38. This PR adds and uses the `ChainModel`. This model exposes chain data, like block heights and hashes. Can be used as a source for list views and repeaters. The model uses a lazy load approach, ie, data is fetched from `interfaces::Chain` as needed.

### CODE REVIEW

File: src/qml/chainmodel.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+bool ChainModel::canFetchMore(const QModelIndex&) const
+{
+    return d->blocks.size() == 0 || d->blocks[0].height > 0;
+}
+
+void ChainModel::fetchMore(const QModelIndex& parent)
+{
+    auto& chain = Engine::chain(this);
```
Comment (promag) [inline src/qml/chainmodel.cpp:70]: Add `qDebug() << Q_FUNC_INFO;` while testing (scrolling the list) to see how lazy loading works.

File: src/qml/pages/stub.qml [hunk 2]
```diff
@@ -14,12 +15,38 @@ ApplicationWindow {
...
+            BitcoinCoreComponents.BlockCounter {
+                id: blockCounter
+                anchors.centerIn: parent
+                height: parent.height / 3
+                blockHeight: node_model.blockTipHeight
+            }
+
+            ListView {
+                anchors.fill: parent
+                model: ChainModel {
```
Comment (promag) [inline src/qml/pages/stub.qml:38]: @ryanofsky this is a dumb example, but the idea is to access block details.

File: src/qml/chainmodel.cpp (hunk 1 above)
Comment (promag) [reply src/qml/chainmodel.cpp:70]: @ryanofsky using `interfaces::Chain` here.

**Comment by jarolrod** [comment]
Concept ACK, we're going to need to expose more than just the nodemodel to qml. One big backend object would be quite a mess. All that's needed is good documentation of what is exposed to qml and why.

**Pull request closed without merging**